test/swap_test.ts

```diff
@@ -1,39 +1,303 @@
 
-const SUITE_NAME = 'swap tests';
+const SUITE_NAME = '/swap';
+
+const DEFAULT_QUERY_PARAMS = {
+    buyToken: 'ZRX',
+    sellToken: 'WETH',
+    excludedSources: 'Uniswap,Eth2Dai,Kyber,LiquidityProvider',
+};
+
+const ONE_THOUSAND_IN_BASE = new BigNumber('1000000000000000000000');
 
 describe(SUITE_NAME, () => {
+    let meshUtils: MeshTestUtils;
+    let accounts: string[];
+    let takerAddress: string;
+    let makerAddress: string;
+
+    let blockchainLifecycle: BlockchainLifecycle;
+    let provider: Web3ProviderEngine;
+
     before(async () => {
         await setupApiAsync(SUITE_NAME);
+        // connect to ganache and run contract migrations
+        const ganacheConfigs = {
+            shouldUseInProcessGanache: false,
+            shouldAllowUnlimitedContractSize: true,
+            rpcUrl: config.ETHEREUM_RPC_URL,
+        };
+        provider = web3Factory.getRpcProvider(ganacheConfigs);
+
+        const web3Wrapper = new Web3Wrapper(provider);
+        blockchainLifecycle = new BlockchainLifecycle(web3Wrapper);
+
+        accounts = await web3Wrapper.getAvailableAddressesAsync();
+        [makerAddress, takerAddress] = accounts;
+
+        // Set up liquidity.
+        await blockchainLifecycle.startAsync();
+        await setupMeshAsync(SUITE_NAME);
+        meshUtils = new MeshTestUtils(provider);
+        await meshUtils.setupUtilsAsync();
+        await meshUtils.addPartialOrdersAsync([
+            {
+                makerAssetData: ZRX_ASSET_DATA,
+                takerAssetData: WETH_ASSET_DATA,
+                makerAssetAmount: ONE_THOUSAND_IN_BASE,
+                takerAssetAmount: ONE_THOUSAND_IN_BASE,
+            },
+            {
+                makerAssetData: ZRX_ASSET_DATA,
+                takerAssetData: WETH_ASSET_DATA,
+                makerAssetAmount: ONE_THOUSAND_IN_BASE,
+                // tslint:disable:custom-no-magic-numbers
+                takerAssetAmount: ONE_THOUSAND_IN_BASE.multipliedBy(2),
+            },
+            {
+                makerAssetData: ZRX_ASSET_DATA,
+                takerAssetData: WETH_ASSET_DATA,
+                makerAssetAmount: MAX_MINT_AMOUNT,
+                // tslint:disable:custom-no-magic-numbers
+                takerAssetAmount: ONE_THOUSAND_IN_BASE.multipliedBy(3),
+            },
+            {
+                makerAssetData: WETH_ASSET_DATA,
+                takerAssetData: ZRX_ASSET_DATA,
+                makerAssetAmount: MAKER_WETH_AMOUNT,
+                takerAssetAmount: ONE_THOUSAND_IN_BASE,
+            },
+        ]);
     });
     after(async () => {
+        await blockchainLifecycle.revertAsync();
+        await teardownMeshAsync(SUITE_NAME);
         await teardownApiAsync(SUITE_NAME);
     });
-
-    describe('should respond to GET /swap/quote', () => {
+    describe('/quote', () => {
         it("with INSUFFICIENT_ASSET_LIQUIDITY when there's no liquidity (empty orderbook, sampling excluded, no RFQ)", async () => {
```

Comment:
nit: can we change to "should response with..." for readability

Reply:
well this is describing the end-point its testing.